pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -404,19 +404,39 @@ public synchronized PinotResourceManagerResponse addInstance(@Nonnull Instance i
     TableType tableType = TableNameBuilder.getTableTypeFromTableName(tableNameWithType);
     List<String> segmentNames = new ArrayList<>();
     if (tableType == TableType.OFFLINE) {
-      for (OfflineSegmentZKMetadata segmentZKMetadata : ZKMetadataProvider.getOfflineSegmentZKMetadataListForTable(
-          _propertyStore, tableNameWithType)) {
+      for (OfflineSegmentZKMetadata segmentZKMetadata : getOfflineSegmentMetadata(tableNameWithType)) {
```

Comment:
Here we don't need the extra table name or type check right? We should keep the new method, but here use the old way to fetch segment metadata.